lib/ui/common/roundedButton.dart

```diff
@@ -2,125 +2,156 @@ import 'package:flutter/material.dart';
-
   final BoxShape buttonShape;
   final BorderRadius borderRadius;
-
   final EdgeInsets edgePadding;
   final TextStyle buttonTextStyle;
-
-  final double size;
+  final double height;
+  final double width;
   final double iconEdgePadding;
   final double buttonIconSize;
 
-  RoundedButtonStyle(this.size,
-      this.iconButtonColor, this.backgroundColor, this.iconEdgePadding,
-      {this.buttonShape, this.borderRadius, this.edgePadding,
-        this.buttonTextStyle, this.buttonIconSize});
-}
-
-class _DefaultStyle implements RoundedButtonStyle {
-
-  final Color iconButtonColor =  const Color(0xFFFFFFFF);
-  final Color backgroundColor =  const Color(0xff24d900);
-
-  final BoxShape buttonShape = BoxShape.rectangle;
-  final BorderRadius borderRadius = const BorderRadius.all(Radius.circular(20));
-
-  final EdgeInsets edgePadding = const EdgeInsets.only(left: 32, top: 31, right: 32, bottom: 32);
-  final TextStyle buttonTextStyle = const TextStyle(fontSize: 17, fontWeight: FontWeight.bold, color: Color(0xffffffff));
+  RoundedButtonStyle({
+    this.height,
+    this.width,
+    this.iconButtonColor,
+    this.backgroundColor,
+    this.iconEdgePadding,
+    this.buttonShape,
+    this.borderRadius,
+    this.edgePadding,
+    this.buttonTextStyle,
+    this.buttonIconSize,
+  });
+
+  factory RoundedButtonStyle.defaultStyle() {
+    return RoundedButtonStyle(
+      iconButtonColor: Color(0xFFFFFFFF),
+      backgroundColor: Color(0xff24d900),
+      buttonShape: BoxShape.rectangle,
+      borderRadius: BorderRadius.all(Radius.circular(20)),
+      edgePadding: EdgeInsets.all(0),
+      buttonTextStyle: null,
+      height: 24.0,
+      width: 24.0,
+      iconEdgePadding: 0,
+      buttonIconSize: 9.0,
+    );
+  }
 
-  final double iconEdgePadding = 5;
-  final double size = 56.0;
-  final double buttonIconSize = null;
+  factory RoundedButtonStyle.bigRoundedButton() {
+    return RoundedButtonStyle.defaultStyle().copyWith(
+        borderRadius: BorderRadius.all(Radius.circular(40)),
+        height: 48.0,
+        width: 48.0,
+        buttonIconSize: 14.0);
+  }
 
-  const _DefaultStyle();
-}
+  factory RoundedButtonStyle.largeRoundedButtonStyle() {
+    return RoundedButtonStyle.defaultStyle().copyWith(
+      borderRadius: BorderRadius.all(Radius.circular(16)),
+      edgePadding: EdgeInsets.only(left: 32, top: 31, right: 32, bottom: 32),
+      buttonTextStyle: TextStyle(
+          fontSize: 17, fontWeight: FontWeight.bold, color: Color(0xffffffff)),
+      iconEdgePadding: 5,
+      height: 56,
+      width: double.infinity,
+      buttonIconSize: null,
+    );
+  }
 
-class RoundedButton {
-  static build(
-      {RoundedButtonStyle style = const _DefaultStyle(),
-        @required  BuildContext context,
-        String title, IconData icon,
-        Function onTap}) {
-    return _buildRoundedButton(style, context , title: title, icon: icon, onTap: onTap);
+  factory RoundedButtonStyle.actionSheetLargeRoundedButton() {
```

Comment:
Add a TODO comment to refactor this on another PR style refactor. This should be outside of this class should only have the default one